server/api/log.go

```diff
@@ -69,3 +71,52 @@ func (h *logHandler) Handle(w http.ResponseWriter, r *http.Request) {
 
 	h.rd.JSON(w, http.StatusOK, "The log level is updated.")
 }
+
+const defaultGetLogSecond = 600 // 10 minutes
+
+// @Tags admin
+// @Summary Get logs.
+// @Param name query []string true "name" collectionFormat(multi)
+// @Param second query integer false "duration of getting"
+// @Produce plain
+// @Success 200 {string} string "Finished getting logs."
+// @Failure 400 {string} string "The input is invalid."
+// @Router /admin/log [get]
+func (h *logHandler) GetLog(w http.ResponseWriter, r *http.Request) {
+	query := r.URL.Query()
+	var second int64
+	if secondStr := query.Get("second"); secondStr != "" {
+		var err error
+		second, err = strconv.ParseInt(secondStr, 10, 64)
+		if err != nil {
+			h.rd.JSON(w, http.StatusBadRequest, err.Error())
+			return
+		}
+	}
+	if second <= 0 {
+		second = defaultGetLogSecond
+	}
+
+	names := query["name"]
+	if len(names) == 0 {
+		h.rd.JSON(w, http.StatusBadRequest, "empty name.")
+		return
+	}
+
+	logConfig := h.svr.GetConfig().Log
+	logConfig.Format = "json"
+	logConfig.Level = "debug"
+	httpLogger, err := logutil.NewHTTPLogger(&logConfig, w)
+	if err != nil {
+		h.rd.JSON(w, http.StatusBadRequest, err.Error())
+		return
+	}
+
+	httpLogger.Plug(names...)
```

Comment:
If there is no logger for a given name, we should return the message to the client?

Reply:
good idea